File: C++/389. Find the Difference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        int i;
        char a;
        sort(s.begin(),s.end());
        sort(t.begin(),t.end());
        if(s.size()>t.size())
        {
            for(i=0;i<t.size();i++)
            {
                if(t[i]!=s[i])
                {a=s[i];
                break;}
            }
            if(a==NULL)
            a=s[s.size()-1];
        }
        else
        {
            for(i=0;i<s.size();i++)
            {
                if(s[i]!=t[i])
                {a=t[i];
                break;}
            }
            if(a==NULL)
            a=t[t.size()-1];
        }
        return a;
    }
};

File: JAVA/389. Find the Difference.java
class Solution {
    public char findTheDifference(String s, String t) {
        char a[]=s.toCharArray();
        char b[]=t.toCharArray();
        Arrays.sort(a);
        Arrays.sort(b);
        char n='\0';
        if(s.length()>t.length())
        {
            for(int i=0;i<t.length();i++)
            {
                if(a[i]!=b[i])
                {
                    n=a[i];
                    break;
                }
            }
            if(n=='\0')
            n=a[a.length-1];
        }
        else
        {
            for(int i=0;i<s.length();i++)
            {
                if(a[i]!=b[i])
                {
                    n=b[i];
                    break;
                }
            }
            if(n=='\0')
            n=b[b.length-1];
        }
        return n;
    }
}